meadow/html/index.html

```diff
@@ -53,1134 +67,1426 @@
-</script>
-     <div id="tabs">
-      <ul>
-        <li><a draggable="false" href="#tabs-1">Smart interface</a></li>
-	<li><a draggable="false" href="#tabs-2">Manual interface</a></li>
-      </ul>
-      <div id="tabs-1">
-        <table style="border:1px solid black; width:100%">
-	  <thead>
-	  <tr>
-	    <th width="100%"><b>Potential actions</b></th><th align="center" valign="middle"><button onclick="Haste.refreshActions()">&nbsp;&nbsp;&nbsp;Refresh&nbsp;&nbsp;&nbsp;</button></th>
-	  </tr>
-	  </thead>
-	  <tbody id="actions">
-	  <tr>
-	    <td align="center" valign="middle">Loading...</td><td></td>
-	  </tr>
-	  </tbody>
-        </table>
-      </div>
-      <div id="tabs-2">
+  </head>
+  <body>
+    <div id="main">
+      <div id="blocklyArea"></div>
+      <div id="ui">
+        <div id="codeAreaDiv"></div>
+        <script>
+          codeArea = CodeMirror(document.getElementById('codeAreaDiv'), {
+            value:
+              'CommitCash (IdentCC 1) 1\n           (ConstMoney 100)\n           10 200\n           (CommitCash (IdentCC 2) 2\n                       (ConstMoney 20)\n                       20 200\n                       (When (PersonChoseSomething (IdentChoice 1) 1)\n                             100\n                             (Both (RedeemCC (IdentCC 1) Null)\n                                   (RedeemCC (IdentCC 2) Null))\n                             (Pay (IdentPay 1) 2 1\n                                  (ConstMoney 20)\n                                  200\n                                  (Both (RedeemCC (IdentCC 1) Null)\n                                        (RedeemCC (IdentCC 2) Null))))\n                       (RedeemCC (IdentCC 1) Null))\n           Null',
+            mode: 'text/x-haskell'
+          });
+          codeArea.setSize('100%', '100%');
+        </script>
+        <div id="actionButtons">
+          <div class="row1">
+            <button id="b2c" onclick="Haste.b2c()">-&gt; Blockly to Code</button>
+            <button id="c2b" onclick="Haste.c2b()">Code to Blockly &lt;-</button>
+            <button id="clear" onclick="Haste.clear_workspace()">Clear</button>
+            <button id="execute" onclick="Haste.execute()">Execute</button>
+          </div>
+          <div class="row2">
+            <button
+              id="embedding"
+              onclick="ifr=document.getElementById('iframe'); if (ifr == null) {var ifr=document.createElement('iframe');ifr.setAttribute('id', 'iframe');document.body.appendChild(ifr);ifr.setAttribute('src', 'editor/index.html');} else { ifr.style.display = 'block'; }"
+            >
+              Use Haskell embedding editor (Fay)
+            </button>
+          </div>
+        </div>
+
+        <br /><br />
+
+        <div class="group">
+          <h3 style="display:inline;" >Current block:</h3>
+	  <input style="display:inline;" type="number" id="currBlock" value="0" />
+<!--      <span class="highlight"></span>
```

Comment:
I have inlined the current block spinbox, but it seems I broke the highlighting in the process, so I commented it out. Is there a way we can have both things?